### ISSUE TITLE
import_validator doesn't map PyPI package names to import names (pyyaml→yaml, Pillow→PIL, etc.)

### ISSUE THREAD

**Issue by martymcenroe** [martymcenroe/AssemblyZero#1518]
## Symptom (verified)

Chiron #37 iter10 (commit `b0eac95a3` + #1515/#1516 from PR #1517, with Chiron PR #40's `tests/fixtures/sample-mpep.pdf` committed and the iter08 orphan branch renamed out of the way). LLD APPROVED in one pass. Impl stage proceeded through several files, then halted:

```
FATAL: Failed to implement src/chiron/corpus/cli.py: Validation failed
after 2 attempts: Unresolvable imports: yaml (line 9)
```

Chiron's `pyproject.toml` declares `pyyaml>=6.0` in `[project] dependencies`. The PEP 621 reader from #1515 picks up `pyyaml` correctly. But `import yaml` is the canonical import — PyYAML installs as the `yaml` module. The validator's `_normalize_package_name` only does hyphen→underscore normalization, so `pyyaml` stays `pyyaml`, `yaml` is never registered, and the import gets flagged.

## Root cause

`assemblyzero/workflows/testing/nodes/implementation/import_validator.py::_normalize_package_name`:

```python
def _normalize_package_name(name: str) -> str:
    return name.lower().replace("-", "_").split(".")[0]
```

`"pyyaml".lower().replace("-", "_").split(".")[0]` → `"pyyaml"`. The validator then checks `top_level == "yaml"` against the set `{"pyyaml", ...}` and gets False.

This is a classic Python ecosystem mismatch: PyPI publishes under one name, packages install under another. The validator needs a mapping.

## Proposed fix

Two complementary mechanisms:

1. **A hardcoded `_PYPI_TO_IMPORT_NAMES` map** for the common cases that everyone hits — small, deterministic, no runtime cost. Initial set:

   | PyPI name | Import name(s) |
   |---|---|
   | pyyaml | yaml |
   | python-dateutil | dateutil |
   | python-jose | jose |
   | pillow | PIL |
   | beautifulsoup4 | bs4 |
   | opencv-python | cv2 |
   | scikit-learn | sklearn |
   | protobuf | google.protobuf, google |
   | msgpack-python | msgpack |

2. **A runtime fallback via importlib.metadata**: when the package is locally installed (poetry/uv has set up the venv), read `top_level.txt` from the installed dist metadata. Authoritative for any package the local env actually has. Optional — degrades to the hardcoded map if metadata not available.

For Chiron #37 specifically, only the hardcoded map is needed (pyyaml→yaml unblocks).

## Test plan

- `test_normalize_pyyaml_to_yaml`
- `test_normalize_pillow_to_pil` (case-insensitive)
- `test_normalize_beautifulsoup4_to_bs4`
- `test_unmapped_pypi_name_still_normalizes` (regression: `pytest-cov` → `pytest_cov`)
- `test_pyyaml_in_pep621_deps_recognizes_yaml_import` — integration test mirroring Chiron #37 iter10 shape
- Existing tests unchanged.

## Non-claims (per #1390 rule 7)

- Not exhaustive. The mapping seeds the common cases that bite real projects. New entries can be added when surfaced.
- Not claiming importlib.metadata works in every env. The hardcoded map is the floor; metadata is opportunistic.
- Not claiming the LLM-generated code is wrong. `import yaml` is correct Python idiom; the validator is the part out of sync.

## Workaround for Chiron #37 iter11

None applied. The fix is the unblocker.
